File: 002-mockowanie/Java/src/test/java/lukaszstaniszewski/service/CreateOrderServiceTest.java
package lukaszstaniszewski.service;

import lukaszstaniszewski.util.EmailInterface;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.runners.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class CreateOrderServiceTest {

    @Mock
    private EmailInterface emailInterface;

    private CreateOrderService createOrderService;

    @Before
    public void setUp() throws Exception  {
        this.createOrderService = new CreateOrderService(this.emailInterface);
    }

    @Test
    public void successCreateOrder() throws Exception {
        Mockito.when(this.emailInterface.send(Mockito.isA(String.class))).thenReturn(true);

        this.createOrderService.createOrder("kontakt@lukaszstaniszewski.pl");

        Mockito.verify(this.emailInterface).send("kontakt@lukaszstaniszewski.pl");
    }

    @Test(expected = Exception.class)
    public void notSendEmail() throws Exception {
        Mockito.when(this.emailInterface.send(Mockito.isA(String.class))).thenReturn(false);

        this.createOrderService.createOrder("kontaktlukaszstaniszewski.pl");

        Mockito.verify(this.emailInterface).send("kontaktlukaszstaniszewski.pl");
    }
}

File: 002-mockowanie/PHP/tests/Service/CreateOrderServiceTest.php
<?php
declare(strict_types=1);

namespace App\Test\Service;

use App\Service\CreateOrderService;
use App\Util\EmailInterface;
use PHPUnit\Framework\TestCase;
use \Mockery;
use \Exception;

/**
 * Class CreateOrderServiceTest
 * @package App\Test\Service
 */
class CreateOrderServiceTest extends TestCase
{
    //Trait for Integration with PHPUnit, second method is use listener in PHPUnit configuration
    use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;

    /**
     * @var
     */
    private $email;

    /**
     * @var CreateOrderService
     */
    private $createOrderService;

    /**
     *
     */
    public function setUp(): void
    {
        $this->email = Mockery::mock(EmailInterface::class);
        $this->createOrderService = new CreateOrderService($this->email);
    }

    /**
     * @throws Exception
     * @test
     */
    public function createOrder(): void
    {
        $this->email->shouldReceive('send')->once()->andReturnTrue();
        $this->createOrderService->createOrder('kontakt@lukaszstaniszewski.pl');
    }

    /**
     * @throws Exception
     * @test
     */
    public function notSendEmail(): void
    {
        $this->email->shouldReceive('send')->once()->andReturnFalse();

        $this->expectException(Exception::class);
        $this->createOrderService->createOrder('kontaktlukaszstaniszewski.pl');
    }
}
